### PR TITLE
stacktrace: Remove dangerous value access

### PR DESCRIPTION
**Pull request by Bojun-Seo**
`dlip.dli_fname` could be `NULL`.
So add null check before use this value.

### CODE REVIEW

**Comment by honggyukim** [comment]
Thanks. I'll have a look later.

**Comment by Bojun-Seo** [comment]
`dladdr` function could be failed. There could be only three cases.
Followings are the way to print for each cases.
Consider symbol name as A, symbol offset as B, shared object name as C and shared object offset as D

Case 1: Success
  A +B (C +D)
Case 2: Failed to get symbol but succeed to get shared object
  C +D
Case 3: Failed to get symbol as well as shared object
  unknown

File: src/stacktrace.cc [hunk 1]
```diff
@@ -95,23 +95,25 @@ static void print_backtrace_symbol(int count, void *addr)
...
 	pr_out("%2d [%#10lx] ", count, (unsigned long)addr);
 #else
 	pr_out("%2d [%#14lx] ", count, (unsigned long)addr);
 #endif
+	// dladdr() translates address to symbolic info.
+	dl_ret = dladdr(addr, &dlip);
+	if (dl_ret == 0) {
+		pr_out("unknown\n");
+		return;
+	}
```
Comment (honggyukim) [inline src/stacktrace.cc:111]: I think printing `?` is better than `unknown` so how about this?
```suggestion
        if (!dladdr(addr, &dlip)) {
                pr_out("?\n");
                return;
        }
```

File: src/stacktrace.cc [hunk 2]
```diff
@@ -95,23 +95,25 @@ static void print_backtrace_symbol(int count, void *addr)
...
-
-	if (status == -2 && !symbol)
-		symbol = strdup(dlip.dli_sname);
-
-	if (symbol) {
-		int len = SYMBOL_MAXLEN;
+	if (dlip.dli_sname != NULL && dlip.dli_saddr != NULL) {
+		symbol = abi::__cxa_demangle(dlip.dli_sname, 0, 0, &status);
+		if (status != 0)
+			symbol = strdup(dlip.dli_sname);
```
Comment (honggyukim) [inline src/stacktrace.cc:116]: I think this change is not needed because we already added return value check of `dladdr` above.

**Review by honggyukim — COMMENTED** [review]
Sorry for the late reply. I've just made a few comments. Thanks for your help!

File: src/stacktrace.cc (hunk 1 above)
Comment (Bojun-Seo) [reply src/stacktrace.cc:111]: `?` seems better, I'll change it as you suggested

**Review by honggyukim — APPROVED** [review]
I've slightly modified your commit changing `NULL` to `nullptr`, then LGTM now.

Thanks for your contribution!

**Pull request merged**